docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
```

Comment:
Does the user really want to share event types (i.e. events he/she defined with his/her authentication info) in Registry ? After applying this new feature, every event a user created has to be shared with others who has access to the namespace. Will it cause any security concern ?

Reply:
That is a good question, and the answer is that I'm not sure :)
You are right, as of now, EventTypes are namespaced-scope and will potentially be consumed from anyone within the namespace.

Reply:
> events he/she defined with his/her authentication info

What does this mean? My reading of the proposal is that the EventTypes themselves have _no_ authentication info inside them. They do imply that some other piece has the authentication info (e.g. something must have authentication info to connect to GitHub's `my-other-user/my-other-repo`). But visibility of the registry itself does not reveal the authentication info, because it doesn't contain any. It may act as a kind of oracle to show what other auth info must exist.

---

In general I think there are two possible security concerns (please correct me if there are others):

1. A user that didn't know about super-secret-event looks in the registry and sees that it exists. 
1. The user now uses that knowledge to create a Trigger to process super-secret-events.

For 2, I don't think this is a problem. Because if the user can create Triggers, they always can create a Trigger that accepts all events. So knowing super-secret-event exists may make things easier, but does not actually expose more information.

For 1, this seems like a fairly narrow case. It involves users that can read EventTypes, but not create Triggers. And for the most part, can't read Triggers (as then they could see a Trigger with a filter for super-secret-event). I haven't thought it all the way through if this leaks any useful information.